boards: arduino: uno_r4: Add PWM mapping

soburi opened the pull request:
Add an `arduino-header-pwm` node to map the PWM channel.

soburi committed:
dts: bindings: pwm: Add Arduino PWM nexus mapping

Add `adruino-header-pwm` map node to describe well Arduino PWM
pin configuration.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi committed:
boards: arduino: uno_r4: Add PWM mapping

Add an `arduino-header-pwm` node to map the PWM channels.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

pillo79 reviewed (commented):
Thanks for the new macros, definitely welcome! On this topic, is there a reason for not using them in the `pwm-map` entries as well? Lines get long, but it would hide the "random" numbers (9 for D3, etc).

soburi committed:
include: zephyr: dt-bindings: gpio: Add `arduino-header-r3` defines

To make the pin names of the Arduino header easier to understand,
we will introduce a macro definition for the pin names.

We will use this to update the definition of `arduino_uno_r4`.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi commented on boards/arduino/uno_r4/arduino_uno_r4_minima.overlay:
```diff
@@ -72,6 +73,23 @@
+			  <14 0 0 &pwm7 0 0 0>,   /* D8 =  P304 = GPT7_A */
+			  <15 0 0 &pwm7 1 0 0>,   /* D9 =  P303 = GPT7_B */
+			  <16 0 0 &pwm3 1 0 0>,   /* D10 = P112 = GPT3_B */
+			  <19 0 0 &pwm3 0 0 0>;   /* D13 = P111 = GPT3_A */
```
Oh, I forgot.
Since it is a good opportunity, I also made comprehensive updates to other parts.

soburi commented on boards/arduino/uno_r4/arduino_uno_r4_wifi.overlay:
```diff
@@ -72,6 +73,25 @@
+			  <16 0 0 &pwm2 0 0 0>,   /* D10 = P103 = GPT2_A */
+			  <17 0 0 &pwm6 0 0 0>,   /* D11 = P411 = GPT6_A */
+			  <18 0 0 &pwm6 1 0 0>,   /* D12 = P410 = GPT6_B */
+			  <19 0 0 &pwm2 1 0 0>;   /* D13 = P102 = GPT2_B */
```
ditto.

pillo79 reviewed (approved):
Nice! Thanks! :+1:
